axum/src/routing/url_params.rs

```diff
@@ -0,0 +1,43 @@
+use crate::util::{ByteStr, PercentDecodedByteStr};
+use http::Extensions;
+use matchit::Params;
+
+pub(crate) enum UrlParams {
+    Params(Vec<(ByteStr, PercentDecodedByteStr)>),
+    InvalidUtf8InPathParam { key: ByteStr },
+}
+
+pub(super) fn insert_url_params(extensions: &mut Extensions, params: Params) {
+    if let Some(UrlParams::InvalidUtf8InPathParam { .. }) = extensions.get() {
+        // nothing to do here since an error was stored earlier
+        return;
+    }
+
+    let params = params
+        .iter()
+        .filter(|(key, _)| !key.starts_with(super::NEST_TAIL_PARAM))
```

Comment:
This seems a bit weird, will this special-cased parameter name go away with the next PR that normalizes path params?

Reply:
This has to do with nested opaque services. Its implemented as a wildcard route like `/path/*NEST_TAIL_PARAM`. So [`NEST_TAIL_PARAM`](https://github.com/tokio-rs/axum/blob/main/axum/src/routing/mod.rs#L101) is reserved by axum and shouldn't pollute the path params the user might see if using `Path<HashMap<String, String>>`.

Reply:
I see. I think it could use being more obviously special in that case, something like `__private__axum_nest_tail_param`.